File: 7. Reverse Integer/7.py
class Solution:
    def reverse(self, x: int) -> int:
        x = str(x)
        r = ""
        for i in range(len(x)):
            if x[i] == '-':
                r += '-'
                r += x[-i-1]
            else:
                if x[-i-1] != '-':
                    r += x[-i-1]
        r = int(r)
        if -2147483648<r<2147483647:
            return r
        else:
            return 0

File: 7. Reverse Integer/7.cs
public class Solution {
    public int Reverse(int x) {
        bool isNegative = x < 0;

        string str = x.ToString().Trim('-');
        char[] charArray = str.ToCharArray();
        Array.Reverse(charArray);
        string reversedStr = new string(charArray);

        if (isNegative)
            reversedStr = "-" + reversedStr;

    try {
            return int.Parse(reversedStr);
        } catch (OverflowException) {
            return 0;
        }
    }
}